server/src/models/products/product.controller.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Patch,
+  Post,
+} from '@nestjs/common';
+import { Product as ProductModel } from '@prisma/client';
+
+import { ProductService } from './product.service';
+import { CreateProductDto, UpdateProductDto } from './product.dto';
+
+@Controller('products')
+export class ProductController {
+  constructor(private readonly productService: ProductService) {}
+
+  @Get()
+  async getProducts(): Promise<ProductModel[]> {
+    return this.productService.products({});
+  }
+
+  @Get(':id')
+  async getProductById(@Param('id') id: string): Promise<ProductModel> {
+    return this.productService.product({ id });
+  }
+
+  @Post()
+  async createProduct(
+    @Body() { ...data }: CreateProductDto,
+  ): Promise<ProductModel> {
+    const { name, brand, imageUrl } = data;
+    return this.productService.createProduct({
+      name,
+      brand,
+      imageUrl,
+    });
+  }
+
+  @Patch(':id')
```

Comment:
I recommend supporting put over patch, we should discuss this in our next meeting.